src/Umbraco.Infrastructure/Serialization/JsonStringInternConverter.cs

```diff
@@ -0,0 +1,15 @@
+using System.Text.Json;
+
+namespace Umbraco.Cms.Infrastructure.Serialization;
+
+/// <summary>
+/// Converts a string to or from JSON, interning the string when reading.
+/// </summary>
+public sealed class JsonStringInternConverter : ReadOnlyJsonConverter<string>
+{
+    /// <inheritdoc />
+    public override string? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        => reader.GetString() is string value
+        ? string.Intern(value)
+        : null;
+}
```

Comment:
Maybe this entire converter should be removed? string.Intern has many downsides: https://github.com/dotnet/runtime/issues/28368#issuecomment-453204870

Reply:
Looking at the code of string.Intern, it still has the same downsides in .NET 8.0. I think the recommendation there to use a custom collection would be better.